File: ironclient/lib/models/hostedinvitation.dart
import 'package:ironcirclesapp/models/circle.dart';
import 'package:ironcirclesapp/models/hostedfurnace.dart';
import 'package:ironcirclesapp/models/user.dart';

class HostedInvitation {
  String link;
  User inviter;
  Circle? circle;
  HostedFurnace hostedFurnace;

  HostedInvitation({
    required this.link,
    required this.inviter,
    required this.circle,
    required this.hostedFurnace,
  });

  factory HostedInvitation.fromJson(Map<String, dynamic> json) =>
      HostedInvitation(
        link: json['link'],
        inviter: User.fromJson(json["inviter"]),
        circle: json["circle"] == null ? null : Circle.fromJson(json["circle"]),
        hostedFurnace: HostedFurnace.fromJson(json["hostedFurnace"]),
      );

  /*Map<String, dynamic> toJson() => {
        'token': token,
        'inviter': inviter,
        'circle': circle,
        'hostedFurnace': hostedFurnace,
      };

   */
}

class HostedInvitationCollection {
  final List<HostedInvitation> objects;

  HostedInvitationCollection.fromJSON(Map<String, dynamic> json, String key)
      : objects = (json[key] as List)
            .map((json) => HostedInvitation.fromJson(json))
            .toList();
}

File: ironapi/models/hostedinvitation.js
const mongoose = require('mongoose');

var HostedInvitationSchema = new mongoose.Schema({
  token: { type: String, required: true },
  active: { type: Boolean, default: true },
  inviter: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  circle: { type: mongoose.Schema.Types.ObjectId, ref: 'Circle' },
  hostedFurnace: { type: mongoose.Schema.Types.ObjectId, ref: 'HostedFurnace' },
}, { timestamps: { createdAt: 'created', updatedAt: 'lastUpdate' } }, { collection: 'hostedinvitation' });

HostedInvitationSchema.statics.new = async function (json) {

  let object = this(json);
  return object;
}


mongoose.model('HostedInvitation', HostedInvitationSchema);

module.exports = mongoose.model('HostedInvitation');